Warnings compiling current master

Few new warnings since the merges last night
```
  CXX      libbitcoin_server_a-init.o
In file included from init.cpp:22:
./txdb.h:19:1: warning: class 'CDiskTxPos' was previously declared as a struct [-Wmismatched-tags]
class CDiskTxPos;
^
./main.h:209:8: note: previous use is here
struct CDiskTxPos : public CDiskBlockPos
       ^
./txdb.h:19:1: note: did you mean struct here?
class CDiskTxPos;
^~~~~
struct
In file included from init.cpp:26:
./validationinterface.h:12:1: warning: class 'CBlockLocator' was previously declared as a struct [-Wmismatched-tags]
class CBlockLocator;
^
./primitives/block.h:140:8: note: previous use is here
struct CBlockLocator
       ^
./validationinterface.h:12:1: note: did you mean struct here?
class CBlockLocator;
^~~~~
struct
2 warnings generated.
  CXX      libbitcoin_server_a-leveldbwrapper.o
```

```
  CXX      test/test_test_bitcoin-hash_tests.o
In file included from test/hash_tests.cpp:11:
In file included from /usr/local/include/boost/test/unit_test.hpp:19:
/usr/local/include/boost/test/test_tools.hpp:536:17: warning: comparison of integers of different signs: 'const unsigned int' and
      'const int' [-Wsign-compare]
    return left == right;
           ~~~~ ^  ~~~~~
/usr/local/include/boost/test/test_tools.hpp:560:16: note: in instantiation of function template specialization
      'boost::test_tools::tt_detail::equal_impl<unsigned int, int>' requested here
        return equal_impl( left, right );
               ^
/usr/local/include/boost/test/test_tools.hpp:575:16: note: in instantiation of function template specialization
      'boost::test_tools::tt_detail::equal_impl_frwd::call_impl<unsigned int, int>' requested here
        return call_impl( left, right, left_is_array() );
               ^
/usr/local/include/boost/test/test_tools.hpp:523:50: note: in instantiation of function template specialization
      'boost::test_tools::tt_detail::equal_impl_frwd::operator()<unsigned int, int>' requested here
BOOST_PP_REPEAT( BOOST_TEST_MAX_PREDICATE_ARITY, IMPL_FRWD, _ )
                                                 ^
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:38:60: note: expanded from macro 'BOOST_PP_REPEAT_1'
# define BOOST_PP_REPEAT_1(c, m, d) BOOST_PP_REPEAT_1_I(c, m, d)
                                                           ^
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:43:63: note: expanded from macro 'BOOST_PP_REPEAT_1_I'
# define BOOST_PP_REPEAT_1_I(c, m, d) BOOST_PP_REPEAT_1_ ## c(m, d)
                                                              ^
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:56:56: note: expanded from macro 'BOOST_PP_REPEAT_1_5'
# define BOOST_PP_REPEAT_1_5(m, d) BOOST_PP_REPEAT_1_4(m, d) m(2, 4, d)
                                                       ^
note: (skipping 1 expansions in backtrace; use -fmacro-backtrace-limit=0 to see all)
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:54:56: note: expanded from macro 'BOOST_PP_REPEAT_1_3'
# define BOOST_PP_REPEAT_1_3(m, d) BOOST_PP_REPEAT_1_2(m, d) m(2, 2, d)
                                                       ^
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:53:62: note: expanded from macro 'BOOST_PP_REPEAT_1_2'
# define BOOST_PP_REPEAT_1_2(m, d) BOOST_PP_REPEAT_1_1(m, d) m(2, 1, d)
                                                             ^
/usr/local/include/boost/test/test_tools.hpp:511:17: note: expanded from macro 'IMPL_FRWD'
    check_impl( P( BOOST_PP_REPEAT_ ## z( BOOST_PP_ADD( n, 1 ), PRED_PARAMS, _ ) ), \
                ^
test/hash_tests.cpp:30:5: note: in instantiation of function template specialization
      'boost::test_tools::tt_detail::check_frwd<boost::test_tools::tt_detail::equal_impl_frwd, unsigned int, int>' requested here
    T(0x00000000, 0x00000000, "");
    ^
test/hash_tests.cpp:20:33: note: expanded from macro 'T'
#define T(expected, seed, data) BOOST_CHECK_EQUAL(MurmurHash3(seed, ParseHex(data)), expected)
                                ^
/usr/local/include/boost/test/test_tools.hpp:169:5: note: expanded from macro 'BOOST_CHECK_EQUAL'
    BOOST_CHECK_WITH_ARGS_IMPL( ::boost::test_tools::tt_detail::equal_impl_frwd(), "", CHECK, CHECK_EQUAL, (L)(R) )
    ^
/usr/local/include/boost/test/test_tools.hpp:102:27: note: expanded from macro 'BOOST_CHECK_WITH_ARGS_IMPL'
    BOOST_TEST_TOOL_IMPL( check_frwd, P, check_descr, TL, CT )      \
                          ^
/usr/local/include/boost/test/test_tools.hpp:77:37: note: expanded from macro 'BOOST_TEST_TOOL_IMPL'
    ::boost::test_tools::tt_detail::func(                               \
                                    ^
1 warning generated.
  CXX      test/test_test_bitcoin-key_tests.o
```
```
In file included from ./test/test_bitcoin.h:4:
./txdb.h:19:1: warning: class 'CDiskTxPos' was previously declared as a struct [-Wmismatched-tags]
class CDiskTxPos;
^
./main.h:209:8: note: previous use is here
struct CDiskTxPos : public CDiskBlockPos
       ^
./txdb.h:19:1: note: did you mean struct here?
class CDiskTxPos;
^~~~~
struct
In file included from test/miner_tests.cpp:13:
In file included from /usr/local/include/boost/test/unit_test.hpp:19:
/usr/local/include/boost/test/test_tools.hpp:536:17: warning: comparison of integers of different signs: 'const unsigned long' and
      'const int' [-Wsign-compare]
    return left == right;
           ~~~~ ^  ~~~~~
/usr/local/include/boost/test/test_tools.hpp:560:16: note: in instantiation of function template specialization
      'boost::test_tools::tt_detail::equal_impl<unsigned long, int>' requested here
        return equal_impl( left, right );
               ^
/usr/local/include/boost/test/test_tools.hpp:575:16: note: in instantiation of function template specialization
      'boost::test_tools::tt_detail::equal_impl_frwd::call_impl<unsigned long, int>' requested here
        return call_impl( left, right, left_is_array() );
               ^
/usr/local/include/boost/test/test_tools.hpp:523:50: note: in instantiation of function template specialization
      'boost::test_tools::tt_detail::equal_impl_frwd::operator()<unsigned long, int>' requested here
BOOST_PP_REPEAT( BOOST_TEST_MAX_PREDICATE_ARITY, IMPL_FRWD, _ )
                                                 ^
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:38:60: note: expanded from macro 'BOOST_PP_REPEAT_1'
# define BOOST_PP_REPEAT_1(c, m, d) BOOST_PP_REPEAT_1_I(c, m, d)
                                                           ^
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:43:63: note: expanded from macro 'BOOST_PP_REPEAT_1_I'
# define BOOST_PP_REPEAT_1_I(c, m, d) BOOST_PP_REPEAT_1_ ## c(m, d)
                                                              ^
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:56:56: note: expanded from macro 'BOOST_PP_REPEAT_1_5'
# define BOOST_PP_REPEAT_1_5(m, d) BOOST_PP_REPEAT_1_4(m, d) m(2, 4, d)
                                                       ^
note: (skipping 1 expansions in backtrace; use -fmacro-backtrace-limit=0 to see all)
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:54:56: note: expanded from macro 'BOOST_PP_REPEAT_1_3'
# define BOOST_PP_REPEAT_1_3(m, d) BOOST_PP_REPEAT_1_2(m, d) m(2, 2, d)
                                                       ^
/usr/local/include/boost/preprocessor/repetition/repeat.hpp:53:62: note: expanded from macro 'BOOST_PP_REPEAT_1_2'
# define BOOST_PP_REPEAT_1_2(m, d) BOOST_PP_REPEAT_1_1(m, d) m(2, 1, d)
                                                             ^
/usr/local/include/boost/test/test_tools.hpp:511:17: note: expanded from macro 'IMPL_FRWD'
    check_impl( P( BOOST_PP_REPEAT_ ## z( BOOST_PP_ADD( n, 1 ), PRED_PARAMS, _ ) ), \
                ^
test/miner_tests.cpp:244:5: note: in instantiation of function template specialization
      'boost::test_tools::tt_detail::check_frwd<boost::test_tools::tt_detail::equal_impl_frwd, unsigned long, int>' requested here
    BOOST_CHECK_EQUAL(pblocktemplate->block.vtx.size(), 1);
    ^
/usr/local/include/boost/test/test_tools.hpp:169:5: note: expanded from macro 'BOOST_CHECK_EQUAL'
    BOOST_CHECK_WITH_ARGS_IMPL( ::boost::test_tools::tt_detail::equal_impl_frwd(), "", CHECK, CHECK_EQUAL, (L)(R) )
    ^
/usr/local/include/boost/test/test_tools.hpp:102:27: note: expanded from macro 'BOOST_CHECK_WITH_ARGS_IMPL'
    BOOST_TEST_TOOL_IMPL( check_frwd, P, check_descr, TL, CT )      \
                          ^
/usr/local/include/boost/test/test_tools.hpp:77:37: note: expanded from macro 'BOOST_TEST_TOOL_IMPL'
    ::boost::test_tools::tt_detail::func(                               \
                                    ^
2 warnings generated.
  CXX      test/test_test_bitcoin-mruset_tests.o
```
```
  CXX      qt/qt_libbitcoinqt_a-csvmodelwriter.o
  CXX      qt/qt_libbitcoinqt_a-guiutil.o
In file included from qt/guiutil.cpp:41:
In file included from /usr/local/include/boost/filesystem/detail/utf8_codecvt_facet.hpp:18:
/usr/local/include/boost/detail/utf8_codecvt_facet.hpp:176:17: warning: 'boost::filesystem::detail::utf8_codecvt_facet::do_length' hides
      overloaded virtual function [-Woverloaded-virtual]
    virtual int do_length(
                ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__locale:952:17: note: hidden
      overloaded virtual function 'std::__1::codecvt<wchar_t, char, __mbstate_t>::do_length' declared here: type mismatch at 1st parameter
      ('state_type &' (aka '__mbstate_t &') vs 'const std::mbstate_t &' (aka 'const __mbstate_t &'))
    virtual int do_length(state_type&, const extern_type* __frm, const extern_type* __end, size_t __mx) const;
                ^
1 warning generated.
```

--- a/src/txdb.h
+++ b/src/txdb.h
@@ -16,7 +16,7 @@
 
 class CBlockFileInfo;
 class CBlockIndex;
-class CDiskTxPos;
+struct CDiskTxPos;
 class uint256;
 
 //! -dbcache default (MiB)

--- a/src/validationinterface.h
+++ b/src/validationinterface.h
@@ -9,7 +9,7 @@
 #include <boost/signals2/signal.hpp>
 
 class CBlock;
-class CBlockLocator;
+struct CBlockLocator;
 class CTransaction;
 class CValidationInterface;
 class CValidationState;

